core/src/main/scala/scalaz/Memo.scala

```diff
@@ -64,23 +64,25 @@ object Memo extends MemoInstances {
     */
   def doubleArrayMemo(n: Int, sentinel: Double = 0d): Memo[Int, Double] = new DoubleArrayMemo(n, sentinel)
 
+  @deprecated("7.2.19", "removed in scalaz 7.3: leaks mutable state, not thread safe")
   def mutableMapMemo[K, V](a: collection.mutable.Map[K, V]): Memo[K, V] =
     memo[K, V](f => k => a.getOrElseUpdate(k, f(k)))
 
   /** Cache results in a [[scala.collection.mutable.HashMap]].
     * Nonsensical if `K` lacks a meaningful `hashCode` and
     * `java.lang.Object.equals`.
     */
+  @deprecated("7.2.19", "removed in scalaz 7.3: not thread safe, relies on Object")
   def mutableHashMapMemo[K, V]: Memo[K, V] =
     mutableMapMemo(new collection.mutable.HashMap[K, V])
 
   /** As with `mutableHashMapMemo`, but forget elements according to
     * GC pressure.
     */
+  @deprecated("7.2.19", "removed in scalaz 7.3: not thread safe, relies on Object, not efficient")
```

Comment:
Not really review, but fuck weak references. The GC is not a cache management tool.